core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -98,6 +98,15 @@
      */
     public static final Setting<Integer> MAX_INNER_RESULT_WINDOW_SETTING =
         Setting.intSetting("index.max_inner_result_window", 100, 1, Property.Dynamic, Property.IndexScope);
+
+    /**
+     * Index setting describing the maximum value of allowed `script_fields`that can be retrieved
+     * per search request. The default maximum of 50 is defensive for the reason that retrieving
+     * script fields is a costly operation.
+     */
+    public static final Setting<Integer> MAX_SCRIPT_FIELDS_SETTING =
+        Setting.intSetting("index.max_script_fields", 50, 0, Property.Dynamic, Property.IndexScope);
```

Comment:
note: 50 seems high enough for me, do you agree?

Reply:
I'd even go lower here maybe 32?